src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/CertificateData.cs

```diff
@@ -31,156 +33,75 @@ internal struct CertificateData
     {
         internal struct AlgorithmIdentifier
         {
+            public AlgorithmIdentifier(AlgorithmIdentifierAsn algorithmIdentifier)
+            {
+                AlgorithmId = algorithmIdentifier.Algorithm.Value;
+                Parameters = algorithmIdentifier.Parameters?.ToArray() ?? Array.Empty<byte>();
+            }
+
             internal string AlgorithmId;
             internal byte[] Parameters;
         }
 
+        private CertificateAsn certificate;
         internal byte[] RawData;
         internal byte[] SubjectPublicKeyInfo;
-
-        internal int Version;
-        internal byte[] SerialNumber;
-        internal AlgorithmIdentifier TbsSignature;
         internal X500DistinguishedName Issuer;
-        internal DateTime NotBefore;
-        internal DateTime NotAfter;
         internal X500DistinguishedName Subject;
-        internal AlgorithmIdentifier PublicKeyAlgorithm;
-        internal byte[] PublicKey;
-        internal byte[] IssuerUniqueId;
-        internal byte[] SubjectUniqueId;
         internal List<X509Extension> Extensions;
-        internal AlgorithmIdentifier SignatureAlgorithm;
-        internal byte[] SignatureValue;
-
-        internal CertificateData(byte[] rawData)
-        {
-#if DEBUG
-        try
-        {
-#endif
-            DerSequenceReader reader = new DerSequenceReader(rawData);
 
-            DerSequenceReader tbsCertificate = reader.ReadSequence();
-
-            if (tbsCertificate.PeekTag() == DerSequenceReader.ContextSpecificConstructedTag0)
-            {
-                DerSequenceReader version = tbsCertificate.ReadSequence();
-                Version = version.ReadInteger();
-            }
-            else if (tbsCertificate.PeekTag() != (byte)DerSequenceReader.DerTag.Integer)
-            {
-                throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
-            }
-            else
-            {
-                Version = 0;
-            }
-
-            if (Version < 0 || Version > 2)
-                throw new CryptographicException();
-
-            SerialNumber = tbsCertificate.ReadIntegerBytes();
-
-            DerSequenceReader tbsSignature = tbsCertificate.ReadSequence();
-            TbsSignature.AlgorithmId = tbsSignature.ReadOidAsString();
-            TbsSignature.Parameters = tbsSignature.HasData ? tbsSignature.ReadNextEncodedValue() : Array.Empty<byte>();
-
-            if (tbsSignature.HasData)
-                throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
-
-            Issuer = new X500DistinguishedName(tbsCertificate.ReadNextEncodedValue());
+        internal int Version => certificate.TbsCertificate.Version;
+        
+        internal byte[] SerialNumber => certificate.TbsCertificate.SerialNumber.ToArray();
+        
+        internal AlgorithmIdentifier TbsSignature => new AlgorithmIdentifier(certificate.TbsCertificate.SignatureAlgorithm);
```

Comment:
Are all of these properties used?  I think TbsSignature was only built out to ensure we read the values, now the serializer makes sure that's done.

Reply:
Looks like the remaining ones are used. I will remove this one.